strutil/writer.go

```diff
@@ -0,0 +1,59 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package strutil
+
+import (
+	"bytes"
+)
+
+type LimitedWriter struct {
+	buffer   *bytes.Buffer
+	maxLines int
+	maxBytes int
+	written  int
+}
+
+func NewLimitedWriter(maxLines, maxBytes int) *LimitedWriter {
+	return &LimitedWriter{
+		buffer:   &bytes.Buffer{},
+		maxLines: maxLines,
+		maxBytes: maxBytes,
+	}
+}
+
+func (wr *LimitedWriter) Write(data []byte) (int, error) {
+	sz, err := wr.buffer.Write(data)
+	if err != nil {
+		return 0, err
+	}
+	wr.written += sz
+	if wr.written > wr.maxBytes {
+		data := wr.buffer.Bytes()
+		data = TruncateOutput(data, 0, wr.maxBytes)
```

Comment:
```
data = TruncateOutput(data, wr.maxLines, wr.maxBytes)
```

Reply:
This is intended, in Write we just limit by 2 * number of bytes (the 2x was suggested earlier by Gustavo, so added this). Both line and byte limits are applied only when getting the data in `Bytes()`.